de.metas.procurement.base/src/main/java/de/metas/procurement/base/IPMMProductBL.java

```diff
@@ -45,4 +48,17 @@
 	 */
 	void update(I_PMM_Product pmmProduct);
 
+	/**
+	 * Get the PMM_Product for the date, product, partner and M_HU_PI_Item_Product that fit the given ASI the most.
+	 * This means the attribute instances from the ASI of the PMM Product must be a subset of the given ASI's attribute instances
+	 * 
+	 * @param date
+	 * @param productId
+	 * @param partnerId
+	 * @param huPIPId
+	 * @param asi
+	 * @return
```

Comment:
I chose to use the UC with COALESCE because in case of simple PMM_Product entries with no attributes the seqNo is not needed so it could stay null.
If we need the seqNo for other purposes in the future we will probably need to make it mandatory, but not at the time being.